File: src/main/java/com/example/common/entity/Line.java
package com.example.common.entity;

import javax.persistence.*;

/**
 * Created by Anatol on 11.06.16.
 */
@Entity
@Table(name = "line")
public class Line extends BaseEntity {
    @JoinColumn( nullable = false)
    private Long schemeId;
    @Column(nullable = false)
    private Long xBeginCoordinate;
    @Column(nullable = false)
    private Long yBeginCoordinate;
    @Column(nullable = false)
    private Long xEndCoordinate;
    @Column(nullable = false)
    private Long yEndCoordinate;

    public Long getSchemeId() {
        return schemeId;
    }

    public void setSchemeId(Long schemeId) {
        this.schemeId = schemeId;
    }

    public Long getxBeginCoordinate() {
        return xBeginCoordinate;
    }

    public Long setxBeginCoordinate(Long xBeginCoordinate) {
        this.xBeginCoordinate = xBeginCoordinate;
        return xBeginCoordinate;
    }

    public Long getyBeginCoordinate() {
        return yBeginCoordinate;
    }

    public Long setyBeginCoordinate(Long yBeginCoordinate) {
        this.yBeginCoordinate = yBeginCoordinate;
        return yBeginCoordinate;
    }

    public Long getxEndCoordinate() {
        return xEndCoordinate;
    }

    public Long setxEndCoordinate(Long xEndCoordinate) {
        this.xEndCoordinate = xEndCoordinate;
        return xEndCoordinate;
    }

    public Long getyEndCoordinate() {
        return yEndCoordinate;
    }

    public Long setyEndCoordinate(Long yEndCoordinate) {
        this.yEndCoordinate = yEndCoordinate;
        return yEndCoordinate;
    }
}

File: frontend/src/app/dto/Line.ts
import Scheme from "./Scheme";
export default class Line {
  private id: number;
  private xBeginCoordinate: number;
  private yBeginCoordinate: number;
  private xEndCoordinate: number;
  private yEndCoordinate: number;

  getId():number{
    return this.id;
  }

  setId(value:number){
    this.id=value;
  }

  getXBeginCoordinate():number{
      return this.xBeginCoordinate;
      }

  setXBeginCoordinate(value:number){
      this.xBeginCoordinate=value;
      }

  getYBeginCoordinate():number{
      return this.yBeginCoordinate;
      }

  setYBeginCoordinate(value:number){
      this.yBeginCoordinate=value;
      }

  getXEndCoordinate():number{
      return this.xEndCoordinate;
      }

  setXEndCoordinate(value:number){
      this.xEndCoordinate=value;
      }

  getYEndCoordinate():number{
      return this.yEndCoordinate;
      }

  setYEndCoordinate(value:number){
      this.yEndCoordinate=value;
      }
}
